src/lib.rs

```diff
@@ -131,12 +134,44 @@ pub fn decode_single_frame_compressed(
     high_bit: u16,
     pixel_representation: u16,
 ) -> Result<Box<[u8]>, Error> {
+    let i_buffers = [i_buffer];
+    let dims = [width, height, 1];
+    decode_multi_frame_compressed(
+        &i_buffers,
+        &dims,
+        pi_type,
+        ts_type,
+        samples_per_pixel,
+        bits_allocated,
+        bits_stored,
+        high_bit,
+        pixel_representation,
+    )
+}
+
+/// Decodes a multi frame buffer in GDCM
+pub fn decode_multi_frame_compressed(
+    i_buffers: &[&[u8]],
+    dims: &[u32; 3],
+    pi_type: GDCMPhotometricInterpretation,
+    ts_type: GDCMTransferSyntax,
+    samples_per_pixel: u16,
+    bits_allocated: u16,
+    bits_stored: u16,
+    high_bit: u16,
+    pixel_representation: u16,
+) -> Result<Box<[u8]>, Error> {
+    let i_buffer_lens: Vec<usize> = i_buffers.iter().map(|&fragment| fragment.len()).collect();
```

Comment:
Given the auto-referencing rules, I suspect that the `&` isn't needed here, nor in the other closure below.

```suggestion
    let i_buffer_lens: Vec<usize> = i_buffers.iter().map(|fragment| fragment.len()).collect();
```